processor/filterprocessor/config.go

```diff
@@ -266,6 +273,17 @@ var _ component.Config = (*Config)(nil)
 
 // Validate checks if the processor configuration is valid
 func (cfg *Config) Validate() error {
+	// TODO: The old format should be deprecated and removed.
+	if cfg.TraceConditions != nil && (cfg.Traces.SpanConditions != nil || cfg.Traces.SpanEventConditions != nil) {
+		return errors.New("cannot use inferred context trace conditions and old format at the same time")
+	}
+	if cfg.MetricConditions != nil && (cfg.Metrics.MetricConditions != nil || cfg.Metrics.DataPointConditions != nil) {
```

Comment:
I'm trying to wrap my head around on how this processor config should work, but based on the existing settings and validations (docs is missing), it should use either the `include` and  `exclude` settings, OR the OTTL conditions, but not both.

That said, I think we're missing a validation here, so it does not allow the `include` and `exclude` keys when `<sinal>_conditions` are configured.